src/ostree/ot-builtin-commit.c

```diff
@@ -823,33 +822,6 @@ ostree_builtin_commit (int argc, char **argv, OstreeCommandInvocation *invocatio
 
       if (!ostree_repo_commit_transaction (repo, &stats, cancellable, error))
         goto out;
-
-      if (!ot_keyfile_get_boolean_with_default (ostree_repo_get_config (repo), "core",
-                                                "change-update-summary", FALSE,
-                                                &change_update_summary, error))
-        goto out;
-
-      /* No need to update it again if we did for each ref change */
-      if (opt_orphan || !change_update_summary)
-        {
-          gboolean commit_update_summary;
-
-          /* The default for this option is FALSE, even for archive repos,
```

Comment:
Maybe it would be good to have some explanation of the default value over in `_ostree_repo_maybe_regenerate_summary` since we're losing this explanation. How about: "The default for this option is FALSE, because of the performance cost and because servers may want to atomically update the summary after a set of operations (auto-update-summary only updates atomically within a transaction)."